data/transactions/logic/TEAL_opcodes.md

```diff
@@ -1560,3 +1560,71 @@ For boxes that exceed 4,096 bytes, consider `box_create`, `box_extract`, and `bo
 | 0 | BlkSeed | []byte |  |
 | 1 | BlkTimestamp | uint64 |  |
 
+
+## ec_add g
+
+- Opcode: 0xe0 {uint8 curve}
+- Stack: ..., A: []byte, B: []byte &rarr; ..., []byte
+- for curve points A and B, return the curve point A + B
+- **Cost**:  BN254g1=310 BN254g2=430 BLS12_381g1=540 BLS12_381g2=750
+- Availability: v9
+
+`EC` Groups:
+
+| Index | Name | Notes |
+| - | ------ | --------- |
+| 0 | BN254g1 | G1 of the BN254 curve. Points encoded as 32 byte X following by 32 byte Y |
+| 1 | BN254g2 | G2 of the BN254 curve. Points encoded as 64 byte X following by 64 byte Y |
+| 2 | BLS12_381g1 | G1 of the BLS 12-381 curve. Points encoded as 48 byte X following by 48 byte Y |
+| 3 | BLS12_381g2 | G2 of the BLS 12-381 curve. Points encoded as 96 byte X following by 48 byte Y |
+
+
+A and B are curve points in affine representation: X concatenated with Y. Fails if A or B is not in G. Does _not_ check if A and B are in the main prime-order subgroup.
```

Comment:
```suggestion
A and B are curve points in affine representation: field element X concatenated with field element Y. 

Encoding of a field element `Z` is as follows.
For base field elements (Fp), `Z` is represented as big-endian number and must be lower than the field modulus p.
For the quadratic field extension (Fp2), `Z` is encoded as the concatenation of the individual encoding of the coefficients. For an Fp2 element of the form `Z = Z0 + Z1 i`, where `i` is a formal quadratic non-residue, the encoding of Z is the concatenation of the encoding of `Z0` and `Z1` in this order. (In particular, it fails is `Z0` or `Z1` is larger or equal to p).

The point at infinity is encoded as `(X,Y) = (0,0)`.
Groups G1 and G2 are denoted addively.

Fails if A or B is not in G. 
(A and/or B are allowed to be the point at infinity.)
Does _not_ check if A and B are in the main prime-order subgroup.
```

taken from https://eips.ethereum.org/EIPS/eip-2537, needed to ensure canonicality.